### PR TITLE
RC-23 Download XML anand saved with url  as name

### PR DESCRIPTION
**Pull request by miltonleung**

### CODE REVIEW

File: RocketCast/DefaultLibraryExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//
```
Comment (QuantumSpark) [inline RocketCast/DefaultLibraryExtensions.swift]: Why did you put this here?

File: RocketCast/DownloadBridge.swift [hunk 2]
```diff
@@ -11,33 +11,52 @@ import Foundation
...
-        let xmlParser = XMLParser(url: nsURL!)
-        
-        //download it
-        //pass the rss data to another object
-        // get the authrfrom that object
-        //get other info from that object (list of episodes, title of podcast)
-        //put the title, episodes, and author into the POdcastModel
-        
+      do {
+        _ = try XMLString!.writeToFile(filePath, atomically: true, encoding: NSUTF8StringEncoding)
```
Comment (QuantumSpark) [inline RocketCast/DownloadBridge.swift]: remove _=

File: RocketCast/DownloadBridge.swift [hunk 3]
```diff
@@ -11,33 +11,52 @@ import Foundation
...
 }
 extension ModelBridge: DownloadBridgeProtocol {
+  
+  func downloadPodcastXML(url:PodcastWebURL, result:(url: PodcastStorageURL) -> ()) {
+    let podcastURL = NSURL(string: url)
+    
+    let task = NSURLSession.sharedSession().dataTaskWithURL(podcastURL!) {(data, response, error) in
+      
+      guard error == nil else {
+        print(error)
```
Comment (QuantumSpark) [inline RocketCast/DownloadBridge.swift]: use Log.error

File: RocketCast/DownloadBridge.swift [hunk 4]
```diff
@@ -11,33 +11,52 @@ import Foundation
...
-        // get the authrfrom that object
-        //get other info from that object (list of episodes, title of podcast)
-        //put the title, episodes, and author into the POdcastModel
-        
+      do {
+        _ = try XMLString!.writeToFile(filePath, atomically: true, encoding: NSUTF8StringEncoding)
+        result(url: filePathAppend)
+      } catch let error as NSError {
+        print(error.description)
+      }
```
Comment (QuantumSpark) [inline RocketCast/DownloadBridge.swift]: use Log.error

File: RocketCast/PodcastController.swift [hunk 5]
```diff
@@ -16,7 +16,7 @@ class PodcastController: UIViewController {
         super.viewDidLoad()
         setupView()
         ModelBridge.sharedInstance.downloadPodcastXML("http://billburr.libsyn.com/rss") { (downloadedPodcast) in
-            
+            print("AKFHKEHFJAWLJgeregsegsegrF,HBA")
```
Comment (QuantumSpark) [inline RocketCast/PodcastController.swift]: use Log.info

File: RocketCastTests/DownloadTests.swift [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  func testPerformanceExample() {
+    // This is an example of a performance test case.
+    self.measureBlock {
+      // Put the code you want to measure the time of here.
+    }
+  }
+  
+  func testDownloadPodcstXML() {
+    
+    var done = false
```
Comment (QuantumSpark) [inline RocketCastTests/DownloadTests.swift]: try to make this a global field inside this class

File: RocketCastTests/DownloadTests.swift [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    let fileMgr = NSFileManager.defaultManager()    
+    
+    ModelBridge.sharedInstance.downloadPodcastXML(url) { (downloadedPodcast) in
+      done = true
+      
+      let path = NSHomeDirectory().stringByAppendingString(downloadedPodcast)
+      print(path)
+      if let data = fileMgr.contentsAtPath(path) {
+        print(NSString(data:data, encoding: NSUTF8StringEncoding)!)
+      }
```
Comment (QuantumSpark) [inline RocketCastTests/DownloadTests.swift]: use log.info

File: RocketCastTests/DownloadTests.swift [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    
+    var done = false
+    let url = "http://billburr.libsyn.com/rss"
+    let fileMgr = NSFileManager.defaultManager()    
+    
+    ModelBridge.sharedInstance.downloadPodcastXML(url) { (downloadedPodcast) in
+      done = true
+      
+      let path = NSHomeDirectory().stringByAppendingString(downloadedPodcast)
+      print(path)
```
Comment (QuantumSpark) [inline RocketCastTests/DownloadTests.swift]: use log.info

File: RocketCastTests/DownloadTests.swift [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    super.tearDown()
+  }
+  
+  func testExample() {
+    // This is an example of a functional test case.
+    // Use XCTAssert and related functions to verify your tests produce the correct results.
+  }
+  
+  func testPerformanceExample() {
```
Comment (QuantumSpark) [inline RocketCastTests/DownloadTests.swift]: is this self-generated? Do we need this?

File: RocketCast/DefaultLibraryExtensions.swift [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+//  RocketCast
+//
+//  Created by Milton Leung on 2016-09-10.
+//  Copyright © 2016 UBCLaunchPad. All rights reserved.
+//
+
+import Foundation
+
+extension String {
+  func stringByRemovingAll(characters: [Character]) -> String {
```
Comment (QuantumSpark) [inline RocketCast/DefaultLibraryExtensions.swift]: put a comment to explain what this is doing

Comment (JonMercer) [inline RocketCast/DefaultLibraryExtensions.swift]: Agree. Not sure what this is add link from SO where you got this?

File: RocketCast/DefaultLibraryExtensions.swift (hunk 1 above)
Comment (miltonleung) [inline RocketCast/DefaultLibraryExtensions.swift]: its by default

File: RocketCast/DefaultLibraryExtensions.swift [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+    return String(self.characters.filter({ !characters.contains($0) }))
+  }
+  
+  func stringByRemovingAll(subStrings: [String]) -> String {
```
Comment (JonMercer) [inline RocketCast/DefaultLibraryExtensions.swift]: Could you add the SO

File: RocketCastTests/DownloadTests.swift (hunk 9 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: self-generated

File: RocketCast/DownloadBridge.swift (hunk 2 above)
Comment (miltonleung) [inline RocketCast/DownloadBridge.swift]: done

File: RocketCast/DownloadBridge.swift (hunk 3 above)
Comment (miltonleung) [inline RocketCast/DownloadBridge.swift]: done

File: RocketCast/DownloadBridge.swift (hunk 4 above)
Comment (miltonleung) [inline RocketCast/DownloadBridge.swift]: done

File: RocketCast/PodcastController.swift (hunk 5 above)
Comment (miltonleung) [inline RocketCast/PodcastController.swift]: removed

File: RocketCastTests/DownloadTests.swift (hunk 7 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: deon

File: RocketCastTests/DownloadTests.swift (hunk 8 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: deon

File: RocketCast/DefaultLibraryExtensions.swift (hunk 11 above)
Comment (mohamedali92) [inline RocketCast/DefaultLibraryExtensions.swift]: Done

File: RocketCastTests/DownloadTests.swift [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+      
+      let path = NSHomeDirectory().stringByAppendingString(downloadedPodcast)
+      
+      if let data = fileMgr.contentsAtPath(path) {
+        Log.info(String(data:data, encoding: NSUTF8StringEncoding)!)
+      }
+    }
+    
+    waitUntil(6) {done}
+    XCTAssert(done, "completeion should be called")
```
Comment (JonMercer) [inline RocketCastTests/DownloadTests.swift]: You're not asserting anything. This is not a test!
- Assert that the XML is a certain size?
- Asserting that the xml downloaded has the word "title" in it (or something that's the same for all XMLs)?
- What about adding a non-rss link? What should happen then?
- What about multiple rss feeds? Perhaps we should try to download the top 10 podcasts in the world and make sure they work?


The goals of testing is to make sure our code is robust.

**Comment by JonMercer** [comment]
General feedback on tests. I'd like to push harder on testing because there's a >50% chance of refactoring this part of code (as we know from RocketPoll)

File: RocketCastTests/DownloadTests.swift (hunk 12 above)
Comment (miltonleung) [inline RocketCastTests/DownloadTests.swift]: Added test cases accordingly

**Pull request merged**